### PR TITLE
Add codecs to be connection independent

### PR DESCRIPTION
**Pull request by purbon**
So we can have real multiline codecs used within the context of logstash-forwarder.

### CODE REVIEW

**Comment by jordansissel** [comment]
I think we can move the complexity to the lumberjack input in Logstash. In 4f8732ce I added an `#accept` method to block and return a new Lumberjack::Connection.

We can update the lumberjack input to use this and clone the codec each time we get a new connection.

**Comment by purbon** [comment]
Ok, will do that.

Thanks for the changes @jordansissel ... at the end the complexity of this should be in the input as is the one dealing with that codec business.

File: lib/lumberjack/server.rb [hunk 1]
```diff
@@ -138,10 +141,10 @@ def header(&block)
       version, frame_type = get.bytes.to_a[0..1]
 
       case frame_type
-        when FRAME_WINDOW; transition(:window_size, 4)
-        when FRAME_DATA; transition(:data_lead, 8)
-        when FRAME_COMPRESSED; transition(:compressed_lead, 4)
-        else; raise "Unknown frame type: #{frame_type}"
+      when FRAME_WINDOW; transition(:window_size, 4)
```
Comment (jordansissel) [inline lib/lumberjack/server.rb]: This ... non-indented 'case/when' style is hard to read, but we'll discuss that in the future when we get together and figure out rubocop styles.

**Comment by jordansissel** [comment]
My style comment was more just an annotation for the future discussion on style - no change required.

**Comment by jordansissel** [comment]
Merging lgtm.

**Pull request closed without merging**